### PR TITLE
Add 'how many units are affected?' to the product page

### PR DESCRIPTION
**Pull request by KyleMacPherson**
https://trello.com/c/rjd0MTN9/203-5-add-how-many-units-are-affected-to-the-product-page

## Description
- Add :affected_units_status, :number_of_affected_units fields to products table.
- Add units affected question on the product form below the counterfeit question.
- Show answer on summary pages
- Add tests

##Screenshots
Screenshot 2020-11-27 at 11 19 57
Screenshot 2020-11-27 at 11 25 37
Screenshot 2020-11-27 at 11 31 03
Screenshot 2020-11-27 at 11 31 17
Screenshot 2020-11-27 at 13 08 07

## Checklist:
- [x] Have you documented your changes in the pull request description?
- [ ] Does the change present any security considerations?
- [ ] Is any gem functionality overloaded? Eg: Devise controller methods being overloaded.
- [ ] Has acceptance criteria been tested by a peer?
- [ ] Has the CHANGELOG been updated? (If change is worth telling users about.)

### General testing (author)
- [ ] Test without JavaScript
- [ ] Test on small screen

### Accessibility testing (author)
- [ ] Works keyboard only
- [ ] Tested with one screen reader
- [ ] Zoom page to 400% - content still visible
- [ ] Disable CSS - does content make sense and appear in a logical order?

### CODE REVIEW

**Comment by nasirkhanbeis** [comment]
spotted couple issues
1. When invalid value such as non-numeric  is submitted, it doesn't throw validation error.Expected to throw error message I think( Toby to confirm)
2. Add/edit product  - select exact and enter 4 , submit form, it returns validation error. suppose to accept and continue.

File: app/decorators/product_decorator.rb [hunk 1]
```diff
@@ -38,4 +40,19 @@ def subcategory_and_category_label
       product_and_category.first
     end
   end
+
+  def units_affected
```
Comment (StupidCodeFactory) [inline app/decorators/product_decorator.rb:44]: we'd usually do this in using `I18n`

File: app/forms/product_form.rb [hunk 2]
```diff
@@ -29,6 +31,8 @@ class ProductForm
   validates :category, presence: true
   validates :subcategory, presence: true
   validates :name, presence: true
+  validates :affected_units_status, inclusion: { in: Product.affected_units_statuses.keys }
+  validates :number_of_affected_units, presence: true, if: -> { affected_units_status == "approx" || affected_units_status == "exact" }
```
Comment (StupidCodeFactory) [inline app/forms/product_form.rb:35]:
```suggestion
  validates :number_of_affected_units, presence: true, if: -> { affected_units_status.inquiry.approx? || affected_units_status.inquiry.exact? }
```

Comment (KyleMacPherson) [reply app/forms/product_form.rb]: I had a look at this but if `affected_units_status` is nil this fails, so stuck with the original.

File: app/controllers/products_controller.rb [hunk 3]
```diff
@@ -64,4 +64,25 @@ def destroy
   def build_breadcrumbs
     @breadcrumbs = build_back_link_to_case || build_breadcrumb_structure
   end
+
+  def attributes_to_update
+    number_of_affected_units = product_params["exact_units"] if product_params["affected_units_status"] == "exact"
+    number_of_affected_units = product_params["approx_units"] if product_params["affected_units_status"] == "approx"
```
Comment (StupidCodeFactory) [inline app/controllers/products_controller.rb:70]: This typically would be handled in the form object, FormObjects have the responsabilty to do the mapping between the params received via form and serialises them into a hash that either and `ActiveRecord` or the service object.

**Review by nasirkhanbeis — APPROVED** [review]
looks good as expected!

**Review by StupidCodeFactory — APPROVED** [review]

**Pull request merged**